### PR TITLE
Impl EthGetTransactionByHashLimited

### LINKED ISSUE #4703: EthGetTransactionByHashLimited

### PR DESCRIPTION
**Pull request by sudo-shashank**
## Summary of changes

Changes introduced in this pull request:

- Impl `Filecoin.EthGetTransactionByHashLimited` and add test
```
| RPC Method                                | Forest | Lotus |
|-------------------------------------------|--------|-------|
| Filecoin.EthGetTransactionByHashLimited   | Valid  | Valid |
```

## Reference issue to close (if applicable)

Closes #4703 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://www.notion.so/chainsafe/Documentation-practices-1a7ec5e5486b4a56b26d9df6b1677a63),
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by lemmih — APPROVED** [review]

File: src/rpc/methods/eth.rs [hunk 1]
```diff
@@ -1955,54 +1955,81 @@ impl RpcMethod<1> for EthGetTransactionByHash {
...
-            // that the from/to addresses of other messages are conversable to 0x-style
-            // addresses. So we just ignore them.
-            //
-            // This should be "fine" as anyone using an "Ethereum-centric" block
-            // explorer shouldn't care about seeing pending messages from native
-            // accounts.
-            if let Ok(eth_tx) = EthTx::from_signed_message(ctx.chain_config().eth_chain_id, smsg) {
-                return Ok(Some(eth_tx.into()));
-            }
+    type Params = (EthHash, Option<ChainEpoch>);
```
Comment (elmattic) [inline src/rpc/methods/eth.rs:1970]: Why did you choose an`Option<ChainEpoch>` and not just a `ChainEpoch`?

**Review by elmattic — CHANGES_REQUESTED** [review]

File: src/rpc/methods/eth.rs (hunk 1 above)
Comment (sudo-shashank) [reply src/rpc/methods/eth.rs:1970]: because `look_back_limit` is of type `Option<i64>` in `search_for_message`
```
forest_filecoin::state_manager::StateManager
impl<DB> StateManager<DB>
pub async fn search_for_message(self: &Arc<Self>, from: Option<Arc<Tipset>>, msg_cid: Cid, look_back_limit: Option<i64>, allow_replaced: Option<bool>) -> Result<Option<(Arc<Tipset>, Receipt)>, Error>
where
    // Bounds from impl:
    DB: Blockstore + Send + Sync + 'static,
```

Comment (elmattic) [reply src/rpc/methods/eth.rs:1970]: Then let's add another test that checks the case where chain epoch is `None`.

File: src/tool/subcommands/api_cmd.rs [hunk 2]
```diff
@@ -1557,7 +1557,9 @@ fn eth_state_tests_with_tipset<DB: Blockstore>(
             tests.push(RpcTest::identity(EthGetTransactionByHash::request((tx
                 .hash
                 .clone(),))?));
-
+            tests.push(RpcTest::identity(EthGetTransactionByHashLimited::request(
```
Comment (ruseinov) [inline src/tool/subcommands/api_cmd.rs:1560]: yeah, I guess we need a test that has `None` epoch passed, as previously mentioned by @elmattic.

Comment (sudo-shashank) [reply src/tool/subcommands/api_cmd.rs:1560]: for `EthGetTransactionByHash` we use `None` 
```
get_eth_transaction_by_hash(ctx, tx_hash, None).await
```
and for `EthGetTransactionByHashLimited` we use 
```
get_eth_transaction_by_hash(ctx, tx_hash, limit).await
```

File: src/rpc/methods/eth.rs (hunk 1 above)
Comment (sudo-shashank) [reply src/rpc/methods/eth.rs:1970]: https://github.com/ChainSafe/forest/pull/4774/files#r1766769807

Comment (elmattic) [reply src/rpc/methods/eth.rs:1970]: You're speaking of testing `get_eth_transaction_by_hash` function which I think is not the same.

Adding the test:

```
            tests.push(RpcTest::identity(EthGetTransactionByHashLimited::request(
                (tx.hash.clone(), None),
            )?));
```

results in:

```
./forest-tool api compare --lotus /ip4/127.0.0.1/tcp/1234/http --forest /ip4/127.0.0.1/tcp/2345/http forest_snapshot_calibnet_2024-09-10_height_1954295.forest.car.zst --filter EthGetTransactionByHashLimited
| RPC Method                                  | Forest            | Lotus |
|---------------------------------------------|-------------------|-------|
| Filecoin.EthGetTransactionByHashLimited (4) | CustomCheckFailed | Valid |
| Filecoin.EthGetTransactionByHashLimited (8) | Valid             | Valid |
Error: Some tests failed
```

**Comment by ruseinov** [comment]
The CI needs to be green, then we're ready to rumble.

File: src/rpc/methods/eth.rs (hunk 1 above)
Comment (sudo-shashank) [reply src/rpc/methods/eth.rs]: changed `limit` type to `ChainEpoch`

**Comment by sudo-shashank** [comment]
@ruseinov and @elmattic this PR is ready for review

**Review by ruseinov — APPROVED** [review]

**Review by elmattic — APPROVED** [review]
